vendor/bat-native-ledger/src/bat/ledger/internal/server/api/endpoints/get_parameters_unittest.cc

```diff
@@ -0,0 +1,290 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/test/task_environment.h"
+#include "bat/ledger/internal/ledger_client_mock.h"
+#include "bat/ledger/internal/ledger_impl_mock.h"
+#include "bat/ledger/internal/server/api/endpoints/get_parameters.h"
+#include "bat/ledger/ledger.h"
+#include "net/http/http_status_code.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+// npm run test -- brave_unit_tests --filter=GetParametersTest.*
+
+using ::testing::_;
+using ::testing::Invoke;
+
+namespace ledger {
+namespace server {
+namespace api {
+
+class GetParametersTest : public testing::Test {
+ private:
+  base::test::TaskEnvironment scoped_task_environment_;
+
+ protected:
+  std::unique_ptr<ledger::MockLedgerClient> mock_ledger_client_;
+  std::unique_ptr<bat_ledger::MockLedgerImpl> mock_ledger_impl_;
+  std::unique_ptr<GetParameters> parameters_;
+
+  GetParametersTest() {
+    mock_ledger_client_ = std::make_unique<ledger::MockLedgerClient>();
+    mock_ledger_impl_ =
+        std::make_unique<bat_ledger::MockLedgerImpl>(mock_ledger_client_.get());
+    parameters_ = std::make_unique<GetParameters>(mock_ledger_impl_.get());
+  }
+};
+
+TEST_F(GetParametersTest, ServerOK) {
+  ON_CALL(*mock_ledger_impl_, LoadURL(_, _, _, _, _, _))
+      .WillByDefault(
+          Invoke([](
+              const std::string& url,
+              const std::vector<std::string>& headers,
+              const std::string& content,
+              const std::string& contentType,
+              const ledger::UrlMethod method,
+              ledger::LoadURLCallback callback) {
+            ledger::UrlResponse response;
+            response.status_code = 200;
+            response.url = url;
+            response.body = R"({
+              "batRate": 0.2476573499489187,
+              "autocontribute": {
+                "choices": [
+                  5,
+                  10,
+                  15
+                ],
+                "defaultChoice": 20
+              },
+              "tips": {
+                "defaultTipChoices": [
+                  1,
+                  10,
+                  100
+                ],
+                "defaultMonthlyChoices": [
+                  5,
+                  10,
+                  15
+                ]
+              }
+            })";
+            callback(response);
+          }));
+
+  parameters_->CreateRequest([&](
```

Comment:
I think `Request` might be a better name, since we aren't just creating the request, we're performing a request and then processing a response.

Reply:
agree, renamed